packages/inventory-compliance/src/Utilities/FilterConfigBuilder.js

```diff
@@ -0,0 +1,191 @@
+    )
+
+    toTextFilterConfig = (item, handler, value) => ({
+        type: conditionalFilterType.text,
+        label: item.label,
+        id: stringToId(item.label),
+        filterValues: {
+            value,
+            [item.event || 'onChange']: (_event, selectedValues) => {
+                handler(stringToId(item.label), selectedValues);
+            }
+        }
+    });
+
+    toCheckboxFilterConfig = (item, handler, value) => ({
+        type: conditionalFilterType.checkbox,
+        label: item.label,
+        id: stringToId(item.label),
+        filterValues: {
+            value,
+            items: item.items,
+            onChange: (_event, selectedValues) => {
+                handler(stringToId(item.label), selectedValues);
+            }
+        }
+    });
+
+    toRadioFilterConfig = (item, handler, value) => ({
+        type: conditionalFilterType.radio,
+        label: item.label,
+        id: stringToId(item.label),
+        filterValues: {
+            value,
+            items: item.items,
+            onChange: (_event, selectedValues) => {
+                handler(stringToId(item.label), selectedValues);
+            }
+        }
+    });
+
+    toFilterConfigItem = (item, handler, value) => {
+        switch (item.type) {
+            case conditionalFilterType.text:
+                return this.toTextFilterConfig(item, handler, value);
+
+            case conditionalFilterType.checkbox:
+                return this.toCheckboxFilterConfig(item, handler, value);
+
+            case conditionalFilterType.radio:
+                return this.toRadioFilterConfig(item, handler, value);
+
+            default:
+                return null;
+        }
+    };
+
+    buildConfiguration = (handler, states, props = {}) => ({
+        ...props,
+        items: this.config.map((item) => (
+            this.toFilterConfigItem(item, handler, states[stringToId(item.label)])
+        ))
+    });
+
+    defaultValueForFilter = (filter) => {
+        switch (filter.type) {
+            case conditionalFilterType.text:
+                return '';
+            case conditionalFilterType.checkbox:
+                return [];
+            default:
+                return;
+        }
+    }
+
+    initialDefaultState = (defaultStates = {}) => {
+        let initialState = {};
+        this.config.forEach((filter) => {
+            const filterStateName = filter.label.replace(' ', '').toLowerCase();
+            initialState[filterStateName] =
+                defaultStates[filterStateName] || this.defaultValueForFilter(filter);
+        });
+        return initialState;
+    }
+
+    categoryLabelForValue = (value) => {
+        const category = this.config.filter((category) => (
+            category.items ? category.items.map((item) => item.value).includes(value) : false
+        ))[0];
+
+        return category ? category.label : value;
+    };
```

Comment:
I'm not sure this is the problem area, but `category` is some object rather than a string. It's causing this warning and probably the behavior I'm seeing:

```
Warning: Encountered two children with the same key, `.$group_Severity_chip_[object Object]`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```

Reply:
I know about this, there is no way I've seen so far to override the keys that the filterships generated.